walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,174 @@
+    
+    return mandatoryAndOptional.isEmpty() ?
+           searchSuitableSlots(mandatoryTimeTable, request.getDuration()) : mandatoryAndOptional;
+  }
+
+  private Collection<TimeRange> searchSuitableSlots(final List<TimeRange> timeTable,
+                                                    final long duration) {
+    Collection<TimeRange> suitableSlots = new LinkedList<>();
+
+    for (TimeRange emptySlot : timeTable) {
+      if (emptySlot.duration() >= duration) {
+        suitableSlots.add(emptySlot);
+      }
+    }
+
+    return suitableSlots;
+  }
+
+  private TimeTablePair registerRelevantEvents(final Collection<Event> events,
+                                               final Collection<String> mandatoryAttendees,
+                                               final Collection<String> optionalAttendees) {
+    /**
+     * The timeTable collection holds TimeRanges which represent empty time slots for
+     * all the people invited
+     */
+    List<TimeRange> timeTable = new ArrayList<>();
+    List<TimeRange> optionalTimeTable = new ArrayList<>();
+    timeTable.add(TimeRange.WHOLE_DAY);
+    optionalTimeTable.add(TimeRange.WHOLE_DAY);
+
+    for (Event event : events) {
+      if (relevantEvent(event, mandatoryAttendees)) {
+        splitTimeTable(timeTable, event.getWhen());
+        splitTimeTable(optionalTimeTable, event.getWhen());
+      }
+
+      if (relevantEvent(event, optionalAttendees)) {
+        splitTimeTable(optionalTimeTable, event.getWhen());
+      }
+    }
+
+    return new TimeTablePair(timeTable, optionalTimeTable);
+  }
+
+  private Boolean relevantEvent(final Event event, final Collection<String> relevantPeople) {
+    for (String person : relevantPeople) {
+      if (event.getAttendees().contains(person)) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  private void splitTimeTable(final List<TimeRange> originalTimeTable,
+                              final TimeRange timeSplitter) {
+    /**
+     * Possible cases:
+     * 1. originalTimeTable:  [--------------------------]
+     *    splitter:                     [-----]
+     *    result:             [---------]     [----------]
+     *
+     * 2. originalTimeTable:  [------]   ...   [---------]
+     *    splitter:                [-------------]
+     *    result:             [----]             [-------]
+     *
+     * 3. originalTimeTable:  [------]   ...  [----------]
+     *    splitter:                     [-----]
+     *    result:             [------]        [----------]
+     *
+     * 4. originalTimeTable:  [------]           [-------]
+     *    splitter:                     [-----]
+     *    result:             [------]           [-------]
+     *
+     * Mention: for this method, it will only matter if the splitter is contained or not
+     */
+    if (originalTimeTable.isEmpty()) {
+      return;
+    }
+    int firstRelevantTimeslotIdx = TimeRange.lowerBound(originalTimeTable, timeSplitter);
```

Comment:
A comment here what exactly you are finding with this would assist reading convenience. Because the variable name "relevant" doesn't really tell the reader what it is. Future readers would not see that you added the function in the same commit and read the accompanying documentation and implementation for it.